### ISSUE TITLE
RSL not working

### ISSUE THREAD

**Issue by CayeFortuna** [FarbstoffRSL/RSL-Helper#72]
**Bug describtion / Fehlerbeschreibung**
First of all, after installing the app, it doesn't find my Raid.exe. So, as the program suggested, I started Riad.exe from the standard path. When the game opens, It shows this error "In C:/Users/User/Documents/RslHelper/Config/RSLHelper.ini kann nicht geschrieben werden." When I click Ok, the game is running normal, but the RSLHelper is all black, And everything that I try to open just pops up some error of some missing file or something like that.

![image](https://user-images.githubusercontent.com/86170872/122656368-415a3b80-d130-11eb-81ce-41005442b1de.png)

![image](https://user-images.githubusercontent.com/86170872/122656379-66e74500-d130-11eb-83b8-c2e0f98eb0ab.png)

![image](https://user-images.githubusercontent.com/86170872/122656391-89795e00-d130-11eb-8ce5-59ff52e0b622.png)

When I try to open something in the sidebar this pops up:

![image](https://user-images.githubusercontent.com/86170872/122656405-939b5c80-d130-11eb-9a94-34c39972bc8f.png)

**HowTo Reproduce / Wie man es reproduziert**

**Expected behavior / Erwartetes verhalten**
 I expected that after the installation, all the files would be in their place.

**System(please complete the following information):**
 EN

**Game Info(please complete the following information):**
Version [e.g. 4.30.0]

**Additional context / Weiterer Infos**
 If my post sounds a bit rude, i'm sorry. English is not my first language so i tried to be more objetive with the text.

**Comment by BlackHawk3000** [comment]
You don't seem to have write permissions in your documents folder, which can't be right. 
Started as admin?
Have you ever deleted the helper completely, all folders, all files and then reinstalled it?

Reinstall should help.

**Comment by CayeFortuna** [comment]
Yes, i  started everything as admin. I deleted everything and reinstalled it but got the same problem. I downloaded an older version of the RSLHelper, the V1.7 B2, and it worked fine. I'm not sure, but it seems that the V1.7 didn't create any files in the documentes folder. As shown in the second printscreen, the RSLHelper was started as Admin.

**Comment by Dwza** [comment]
Are you running the helper as admin? There is a Information about [how to run the helper](https://github.com/FarbstoffRSL/RSL-Helper/wiki/FAQ-DE#wie-starte-ich-den-rsl-helper-permanent-als-admin) in the wiki.

**Comment by CayeFortuna** [comment]
Yes, I opened the installer as admin and the helper as admin. The thing is, the updater is not checking for the files in the documents folder, so it doesn't open properly. I tried the things of the link you sent but didn't work either. Can any of you send the files that you have in the documents folder? I can try to manually put the files there and see if it works. They seem to be the only thing that is missing.

**Comment by BlackHawk3000** [comment]
It is the same file as in your config folder in the current installation directory (Version 1.7).
C:\Users\<yourusername>\AppData\Local\Programs\Common\RSL_Helper_X64\Config

Its a new update out, maybe your problem is solved with the new version.

**Comment by CayeFortuna** [comment]
I downloaded the new version but its happening the same error. I tried to just put the file of the conf folder, but i didn't work.
This is what I have in that file, do you have something different in the new version?

[Raid]
StartWaitTime=1000
[AutoSellSetup]
Rang1Value=-1
Rang2Value=-1
Rang3Value=-1
Rang4Value=-1
Rang5Value=-1
Rang6Value=-1
SPD5Substat=0
SPD6Substat=0
SPDRang5Value=-1
SPDRang6Value=-1
DefFlatArtifact=0
AtkFlatArtifact=0
HpFlatArtifact=0
[Autoklicker]
PlayFinalSound=1
StandardFPSValue=60
[ClanBossSimulator]
BuffApply=1
BuffLose=1

**Comment by Dwza** [comment]
Does the updater has admin permissions too? I had a problem with my updater so i explizit set admin permissions or at least "start as admin" on the RslHelperUpdater.exe what then solved my problem.

**Comment by Dwza** [comment]
```
[ClanBossSimulator]
BuffApply=1
BuffLose=1
LastSetup=789C3360A8613018C5181800D52F2F09

[Autoklicker]
StandardFPSValue=240
PlayFinalSound=1
AutoFPSDetection=1
UserFPSValue=144
ClickWaitTime=300
UseVaultChamps=1

[MercySystem]
um15613092_AncientValue=48
um15613092_VoidValue=106
um15613092_SacredValue=8
um15613092_AncientValueLeg=38
um15613092_VoidValueLeg=116
um15613092_SacredValueLeg=16
um15613092_AncientValueEpic=24
um15613092_VoidValueEpic=16
um15613092_SacredValueEpic=0
um15613092_AncientValueLegLastHero=4270
um15613092_VoidValueLegLastHero=0
um15613092_SacredValueLegLastHero=5560
um15613092_AncientValueEpicLastHero=3250
um15613092_VoidValueEpicLastHero=1920
um15613092_SacredValueEpicLastHero=520

[Raid]
PathRoTaid=X:\Games\Plarium\PlariumPlay\PlariumPlay.exe
StartWaitTime=1000

[AutoSellSetup]
Rang1Value=5
Rang2Value=5
Rang3Value=5
Rang4Value=5
Rang5Value=3
Rang6Value=2
SPD5Substat=1
SPD6Substat=1
SPDRang5Value=-1
SPDRang6Value=-1
DefFlatArtifact=1
AtkFlatArtifact=1
HpFlatArtifact=1
KeepAccBanner=1
AtkFlatAmulett=0
```
Located in: `C:\Users\<username>\Documents\RslHelper\Config`

**Comment by Propapanda666** [comment]
Hallo, ich habe exakt das gleiche Fehlerszenario, auch alles als Admin gestartet, neuste Version etc.pp.

**Comment by Propapanda666** [comment]
> 
> 
> Hallo, ich habe exakt das gleiche Fehlerszenario, auch alles als Admin gestartet, neuste Version etc.pp.

Korrektur, nicht exakt das gleiche Fehlerszenario, die erste Fehlermeldung ist bei mir eine andere - das hatte ich vorhin übersehen, anbei zwei Screenshots

![grafik](https://user-images.githubusercontent.com/86832776/124311778-8dfd3800-db6e-11eb-8f24-f9a5f945a67c.png)

![grafik](https://user-images.githubusercontent.com/86832776/124311959-d6b4f100-db6e-11eb-9dcf-5e69b357c4d0.png)

**Comment by Dwza** [comment]
@Propapanda666 das sieht om erster linie wie ein rechte problem aus. hast du mal komplett deinstalliert, alles andere geschlossen und mit admin rechten neu installiert?

**Comment by CayeFortuna** [comment]
> Latest Helper Version
> 
> ```
> [ClanBossSimulator]
> BuffApply=1
> BuffLose=1
> LastSetup=789C3360A8613018C5181800D52F2F09
> 
> [Autoklicker]
> StandardFPSValue=240
> PlayFinalSound=1
> AutoFPSDetection=1
> UserFPSValue=144
> ClickWaitTime=300
> UseVaultChamps=1
> 
> [MercySystem]
> um15613092_AncientValue=48
> um15613092_VoidValue=106
> um15613092_SacredValue=8
> um15613092_AncientValueLeg=38
> um15613092_VoidValueLeg=116
> um15613092_SacredValueLeg=16
> um15613092_AncientValueEpic=24
> um15613092_VoidValueEpic=16
> um15613092_SacredValueEpic=0
> um15613092_AncientValueLegLastHero=4270
> um15613092_VoidValueLegLastHero=0
> um15613092_SacredValueLegLastHero=5560
> um15613092_AncientValueEpicLastHero=3250
> um15613092_VoidValueEpicLastHero=1920
> um15613092_SacredValueEpicLastHero=520
> 
> [Raid]
> PathRoTaid=X:\Games\Plarium\PlariumPlay\PlariumPlay.exe
> StartWaitTime=1000
> 
> [AutoSellSetup]
> Rang1Value=5
> Rang2Value=5
> Rang3Value=5
> Rang4Value=5
> Rang5Value=3
> Rang6Value=2
> SPD5Substat=1
> SPD6Substat=1
> SPDRang5Value=-1
> SPDRang6Value=-1
> DefFlatArtifact=1
> AtkFlatArtifact=1
> HpFlatArtifact=1
> KeepAccBanner=1
> AtkFlatAmulett=0
> ```
> 
> Located in: `C:\Users\<username>\Documents\RslHelper\Config`

Thank you. Mine is very different. I'll try downloading the newest version again. If it doesn't work, i'll just use the version 1.7. Thanks for your help.

**Comment by CayeFortuna** [comment]
> Latest Helper Version
> 
> ```
> [ClanBossSimulator]
> BuffApply=1
> BuffLose=1
> LastSetup=789C3360A8613018C5181800D52F2F09
> 
> [Autoklicker]
> StandardFPSValue=240
> PlayFinalSound=1
> AutoFPSDetection=1
> UserFPSValue=144
> ClickWaitTime=300
> UseVaultChamps=1
> 
> [MercySystem]
> um15613092_AncientValue=48
> um15613092_VoidValue=106
> um15613092_SacredValue=8
> um15613092_AncientValueLeg=38
> um15613092_VoidValueLeg=116
> um15613092_SacredValueLeg=16
> um15613092_AncientValueEpic=24
> um15613092_VoidValueEpic=16
> um15613092_SacredValueEpic=0
> um15613092_AncientValueLegLastHero=4270
> um15613092_VoidValueLegLastHero=0
> um15613092_SacredValueLegLastHero=5560
> um15613092_AncientValueEpicLastHero=3250
> um15613092_VoidValueEpicLastHero=1920
> um15613092_SacredValueEpicLastHero=520
> 
> [Raid]
> PathRoTaid=X:\Games\Plarium\PlariumPlay\PlariumPlay.exe
> StartWaitTime=1000
> 
> [AutoSellSetup]
> Rang1Value=5
> Rang2Value=5
> Rang3Value=5
> Rang4Value=5
> Rang5Value=3
> Rang6Value=2
> SPD5Substat=1
> SPD6Substat=1
> SPDRang5Value=-1
> SPDRang6Value=-1
> DefFlatArtifact=1
> AtkFlatArtifact=1
> HpFlatArtifact=1
> KeepAccBanner=1
> AtkFlatAmulett=0
> ```
> 
> Located in: `C:\Users\<username>\Documents\RslHelper\Config`

What other files do you have in the Config folder? I'll try to create the files manually. I made the RslHelper.ini, but i think that there's still something missing.

**Comment by Propapanda666** [comment]
> 
> 
> @Propapanda666 das sieht om erster linie wie ein rechte problem aus. hast du mal komplett deinstalliert, alles andere geschlossen und mit admin rechten neu installiert?

Also ich habe mir die Rechte noch mal genauer angeschaut - es gab einige Dateien bei denen war mein Useraccount und nicht der Administrator der Owner - das lag vermutlich daran das ich das Initialsetup mit meinem Useraccount gemacht hatte und nicht als Admin. Daraufhin habe ich die gesamte Installation manuell gelöscht und alles mit dem Administrator Account installiert, also Initial Setup, Updater. Jetzt sind alle Rechte auf Admin gestellt. Und, was soll ich sagen, keine Fehlermeldung mehr. Ich verstehe das zwar nicht, da ich den Helper als Admin starte - er also ohnehin Zugriffsrechte auf die Dateien hatte - aber was solls. Danke nochmal für deine Mühe - und für das Tool! (Vielleicht als Anregung - noch eine Debuglog-Option einhängen.) 

Viele Grüße, 
            Panda

**Comment by Dwza** [comment]
@CayeFortuna you may could not read this because its german but propapanda666 hat a similar issue. he found out that not all files where installed as admin so he deleted everything manually, downloaded the initial installer, installed it as admin, started updated as admin, than ran the programm as admin, now everything works. could you may test this too ? :D

**Comment by CayeFortuna** [comment]
I tested it but unfortunately didn't work too. What I did to make it barely work was turning the windows defender off. I think the problem is that windows defender was blocking the updater to create the files in the documents folder. With the defender off, i manage to open the RSLHelper and Raid itself. It opened, but didn't work properly. I started having the same problem that propapanda666 was having, but even installing as admin and opening as admin didn't work. I'll keep trying to uninstalling and installing each update, but i think the windows defender is the one thats messing it up.
